arkane/statmech.py

```diff
@@ -226,6 +226,11 @@ def load(self, pdep=False, plot=False):
         species object.
         """
         path = self.path
+        directory = os.path.abspath(os.path.dirname(path))
+        if not os.path.isfile(path):
+            path = os.path.join(directory, path)
+
+        ess_list = ['GaussianLog', 'MolproLog', 'OrcaLog', 'QChemLog', 'TeraChemLog', 'Log']
```

Comment:
Would this not be the same as using `_registered_ess_adapters.keys()` (adding in `Log` if need be)? The advantage of this code would be that you do not need to update this list ever when we add new adapters. If we do this you might want to make `_registered_ess_adapters` no longer private (i.e. `registered_ess_adapters`)

Reply:
I agree.
Also, perhaps we don't need this altogether here. I think we can drop all the `any(type(energy).__name__ == ess for ess in ess_list)` conditions (the only usage of `ess_list`): isn't `isinstance(energy, ESSAdapter)` enough to check whether an object is an adapter?

Reply:
Indeed, I have many lines like `if (isinstance(energy, ESSAdapter)` since that works well and I sloppily left in the hardcoded ess_list from previous steps. Thanks for catching that!